src/parent/services/ParentService.js

```diff
@@ -0,0 +1,47 @@
+import configService from '../../shared/ConfigService';
+import { close } from '../actions/parent';
+
+class ParentService {
+
+    constructor(store) {
+        this.store = store;
+        this.onChildClosed = this.onChildClosed.bind(this);
+    }
+
+    onChildClosed({ name }) {
+        this.store.dispatch(close(name));
```

Comment:
> With the persistence of state stuff being in, we don't want to dispatch the close action when the final child window is closed.

Good spot - didn't think of that.

> Looks like we can fix the problem I mentioned in my last comment if we do this instead

I'm not convinced this will solve the problem - it seems that closing a window using Alt+F4/Taskbar doesn't trigger the `beforeunload` event, so the root component is not unmounted which leads to the errors seen above.